integration-templates/hubspot/hubspot-contacts.ts

```diff
@@ -1,12 +1,35 @@
 import type { NangoSync, HubspotContact } from './models';
 
 export default async function fetchData(nango: NangoSync) {
-    const query = `properties=firstname,lastname,email`;
+    const properties = ['firstname', 'lastname', 'email'];
 
-    for await (const records of nango.paginate({ endpoint: '/crm/v3/objects/contacts', params: { query } })) {
-        const mappedRecords = mapHubspotContacts(records);
+    let totalRecords = 0;
 
-        await nango.batchSave(mappedRecords, 'HubspotContact');
+    try {
+        const endpoint = '/crm/v3/objects/contacts';
+        const config = {
+            params: {
+                properties: properties.join(',')
+            },
+            paginate: {
```

Comment:
@hassan254-prog Does the structure of the [paginate parameters in the providers.yaml](https://github.com/NangoHQ/nango/blob/master/packages/shared/providers.yaml#L660) need to adjust to accommodate this? Do these parameters need to be repeated? It appears they are the same.

Reply:
@KhaliqGant, for most providers, they are the same. What prompted me to implement pagination at the script level was Pipedrive, which offers two different pagination types: offset and cursor. The one previously implemented in `providers.yaml` before writing the integration templates was offset, and Bastien advised me not to change that but to define paginations at the script level going forward. We opted for cursor pagination since it provides better performance with large datasets and is more resilient to changes in the dataset. https://pipedrive.readme.io/docs/core-api-concepts-pagination